modules/react/button/lib/PrimaryButtonKing.tsx

```diff
@@ -0,0 +1,423 @@
+  size?: ButtonSizes;
+  /**
+   * The data label of the Button.
+   * Note: not displayed at `small` size
+   */
+  dataLabel?: String;
+  /**
+   * The icon of the Button.
+   * Note: not displayed at `small` size
+   */
+  icon?: CanvasSystemIcon;
+  /**
+   * Button icon positions can either be `start` or `end`.
+   * If no value is provided, it defaults to `start`.
+   * @default 'start'
+   */
+  iconPosition?: IconPositionsNew;
+  /**
+   * If set to `true`, transform the icon's x-axis to mirror the graphic
+   * @default false
+   */
+  shouldMirrorIcon?: boolean;
+  children?: React.ReactNode;
+}
+
+// Button sizes where data labels are enabled
+const dataLabelSizes = ['medium', 'large'];
+
+const StyledPrimaryButton = styled(Button)<StyledType & PrimaryButtonKingProps>(
+  {
+    ...type.levels.subtext.large,
+    cursor: 'pointer',
+    display: 'inline-flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    boxSizing: 'border-box',
+    boxShadow: 'none',
+    outline: 'none',
+    fontWeight: type.properties.fontWeights.bold,
+    lineHeight: 'normal',
+    whiteSpace: 'nowrap',
+    WebkitFontSmoothing: 'antialiased',
+    MozOsxFontSmoothing: 'grayscale',
+    borderRadius: borderRadius.circle,
+    position: 'relative',
+    verticalAlign: 'middle',
+    overflow: 'hidden',
+    border: '1px solid transparent',
+    background: 'red',
+    transition:
+      'box-shadow 120ms linear, border 120ms linear, background-color 120ms linear, color 120ms linear',
+    '&:disabled, &:disabled:active': {
+      cursor: 'default',
+      boxShadow: 'none',
+      opacity: 0.4,
+    },
+
+    '&:hover:active': {transitionDuration: '40ms'},
+
+    '& > *:first-of-type': {
+      marginLeft: '0',
+    },
+
+    '& > *:last-of-type': {
+      marginRight: 0,
+    },
+  },
+  ({size}) => {
+    switch (size) {
+      case 'large':
+        return {
+          ...type.levels.body.small,
+          fontWeight: type.properties.fontWeights.bold,
+          height: '48px',
+          '& > * ': {
+            margin: `0 ${spaceNumbers.xxs / 2}px`,
+          },
+        };
+      case 'medium':
+      default:
+        return {
+          height: space.xl,
+          '& > * ': {
+            margin: `0 ${spaceNumbers.xxs / 2}px`,
+          },
+        };
+      case 'small':
+        return {
+          height: space.l,
+          '& > * ': {
+            margin: `0 ${spaceNumbers.xxxs / 2}px`,
+          },
+        };
+      case 'extraSmall':
+        return {
+          ...type.levels.subtext.medium,
+          fontWeight: type.properties.fontWeights.bold,
+          height: space.m,
+          '& > * ': {
+            margin: `0 ${spaceNumbers.xxxs / 2}px`,
+          },
+        };
+    }
+  },
+  ({grow}) => grow && {width: '100%', maxWidth: '100%'},
+  ({variant, theme}) => {
+    if (!getPrimaryButtonColorsNew(variant, theme)) {
+      return;
+    }
+
+    const colors: ButtonColors = getPrimaryButtonColorsNew(variant, theme);
```

Comment:
I think `getPrimaryButtonColorsNew` should return type `ButtonColors` instead of forcing the assignment of the type here. This code will error here with a large type error if the return type of `getPrimaryButtonColorsNew` doesn't match `ButtonColors` on this line.

If the return type of `getPrimaryButtonColorsNew` is manually set to `ButtonColors`, Typescript will warn you in the `getPrimaryButtonColorsNew` function instead of here. The context of the error will be better and the stack trace won't be so big. Large type error stack traces are wrapped up if localized context cannot be determined.

Reply:
ahhh ok that makes sense! will fix